tests/reftests/extrasource.test

```diff
@@ -583,7 +583,7 @@ The following actions will be performed:
           got      md5=hash
 [ERROR] Failed to get extra source "i-am-a-patch" of multiple-md5.1: Checksum mismatch
 
-OpamSolution.Fetch_fail("Checksum mismatch")
+Checksum mismatch
```

Comment:
this message feels redundant with the ones above

Reply:
In what way ? The lack of information should be the object of a patch for checksum related errors.

Reply:
```
<><> Processing actions <><><><><><><><><><><><><><><><><><><><><><><><><><><><>
[ERROR] multiple-md5.1/i-am-a-patch: Checksum mismatch for file://${BASEDIR}/p.patch:
          expected md5=hash
          got      md5=hash
[ERROR] Failed to get extra source "i-am-a-patch" of multiple-md5.1: Checksum mismatch

Checksum mismatch
```
"checksum mismatch" is displayed three times one after the other.

I believe the difference is that both errors above are displayed as the error happens and the last one is displayed when the actions finished.

Since this PR is no longer just an API improvement, i think it would be nicer to make it so that such redundant error messages are squashed into one